test/test_charms.py

```diff
@@ -96,6 +102,29 @@ def test_missing_agent_state(self):
         self.assertFalse(rv)
 
 
+class TestCharmNovaCompute(unittest.TestCase):
+
+    def setUp(self):
+        self.mock_jujuclient = MagicMock(name='jujuclient')
+        self.mock_juju_state = MagicMock(name='juju_state')
+        self.mock_ui = MagicMock(name='ui')
+        self.mock_config = MagicMock(name='config')
+
+    def test_relations_skip_unavailable(self):
+        """ Skips set relations that aren't defined from
+        placements deployed charms
+        """
+        deployed_charms = [CharmNovaCompute, CharmMysql,
+                           CharmRabbitMQ, CharmNtp,
+                           CharmNovaCloudController]
+        charm_q = CharmQueue(ui=self.mock_ui, config=self.mock_config,
+                             deployed_charms=deployed_charms)
+        valid_relations = charm_q.filter_valid_relations()
+        self.assertNotIn(('nova-compute:nova-ceilometer',
+                          'ceilometer-agent:nova-ceilometer'),
+                         valid_relations)
+
```

Comment:
as discussed on IRC, you want your tests to cover more than this - e.g. have one test check that all the relations you expect to be in valid_relations are actually there. This test passes if filter_valid_relations is just ```lambda : [] ```